pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -221,6 +222,23 @@ func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	var statusMessage string
+	successStatusMessage := "Deployment %s/%s have correct number of AvailableReplicas which is %s"
+	failStatusMessage := "Deployment %s/%s have incorrect number of AvailableReplicas, have %s but should be %s"
+	if found.Status.AvailableReplicas != size {
+		statusMessage = fmt.Sprintf(failStatusMessage, {{ lower .Resource.Kind }}.Name, {{ lower .Resource.Kind }}.Namespace, found.Status.AvailableReplicas, size)
+	} else {
+		statusMessage = fmt.Sprintf(successStatusMessage, {{ lower .Resource.Kind }}.Name, {{ lower .Resource.Kind }}.Namespace, size)
+	}
```

Comment:
Can we initialize `statusMessage` to the successful case by default? that way the if block only needs to see if it is a failure case. 

```golang
successStatusMessage := "Deployment %s/%s have correct number of AvailableReplicas which is %s"
failStatusMessage := "Deployment %s/%s have incorrect number of AvailableReplicas, have %s but should be %s"
statusMessage := fmt.Sprintf(successStatusMessage, {{ lower .Resource.Kind }}.Name, {{ lower .Resource.Kind }}.Namespace, size)

if found.Status.AvailableReplicas != size {
	statusMessage = fmt.Sprintf(failStatusMessage, {{ lower .Resource.Kind }}.Name, {{ lower .Resource.Kind }}.Namespace, found.Status.AvailableReplicas, size)
}
```

I mean it's probably no big deal if we don't but it does remove the extra else. Basically assume it is going to work and if not, handle the failure.

Reply:
We need to change this one. It is not using status Conditions and follows the recommendations.
If we want to add status for this transition then it would be like 
status.conditions.type = Available
status.conditions.status = True
status.conditions.reason = "Resizing"
.status.conditions.Message = "Resizing available replicas of the Deployment %s owned by the custom resource %s from %d to %d "

Therefore if we fails to update the resize then;

status.conditions.type = Available
status.conditions.status = False
status.conditions.reason = "Resizing"
.status.conditions.Message = "Fail to resizing available replicas of the Deployment %s owned by the custom resource %s from %d to %d : (%s) " -> Pass the error faced

Look at : https://github.com/kubernetes-sigs/kubebuilder/compare/master...camilamacedo86:kubebuilder:test-br?expand=1

Also, in the example; https://kpt.dev/reference/schema/crd-status-convention/ ( it will help you understand better its idea )

Reply:
The implementation change and @NikhilSharmaWe are working on the tests. We will need your help on the review when he finishes by sure :-) @jmrodri